core/src/main/java/com/amplifyframework/api/graphql/GraphQLResponse.java

```diff
@@ -35,19 +35,30 @@
 public final class GraphQLResponse<T> {
     private final T data;
     private final List<Error> errors;
+    private final String nextToken;
 
     /**
      * Constructs a wrapper for GraphQL response.
      * @param data response data with user-defined cast type
-     * @param errors list of error responses as defined
-     *               by GraphQL doc
+     * @param errors list of error responses as defined by GraphQL doc
      */
     public GraphQLResponse(@Nullable T data, @Nullable List<Error> errors) {
+        this(data, errors, null);
+    }
+
+    /**
+     * Constructs a wrapper for GraphQL response.
+     * @param data response data with user-defined cast type
+     * @param errors list of error responses as defined by GraphQL doc
+     * @param nextToken token for retrieving the next batch of results if there are any, otherwise null.
+     */
+    public GraphQLResponse(@Nullable T data, @Nullable List<Error> errors, @Nullable String nextToken) {
```

Comment:
The `nextToken` isn't part of a spec'd GraphQL response. Ideally, the `nextToken` would live somewhere in the specification of the `T data`.  Or in other words, ideally, we wouldn't modify `GraphQLResponse` for this, at all.

Reply:
Yep, totally. I came to this same conclusion yesterday after looking a little closer at the response format, and also looking at how this is done on iOS.  I'll be working on this today.